exp/controllers/azuremachinepoolmachine_controller.go

```diff
@@ -346,9 +346,14 @@ func (ampmr *AzureMachinePoolMachineController) reconcileDelete(ctx context.Cont
 }
 
 func newAzureMachinePoolMachineReconciler(scope *scope.MachinePoolMachineScope) azure.Reconciler {
+	scaleSetVMsSvc, err := scalesetvms.NewService(scope)
+	if err != nil {
+		// TODO: handling this differently requires some refactoring
+		panic("failed to create new scalesetvms service")
```

Comment:
Is this refactoring queued up in a forthcoming PR? I'd vote to at least make sure this is ironed out before the release.

Reply:
No, I forgot this was in here and we should resolve it in this PR, just requires some more rewiring.

Reply:
/hold

Pending refactoring of the `// TODO:` comment.

Reply:
/hold cancel

That wasn't too bad, just needed return the error and update an interface and mock.